File: medplum-java/medplum-model/src/main/java/com/fhir/Communication_Payload.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCommunication_Payload.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Communication_Payload {
  Optional<Attachment> contentAttachment();

  Optional<List<Extension>> modifierExtension();

  Optional<Reference> contentReference();

  Optional<List<Extension>> extension();

  Optional<String> contentString();

  Optional<String> id();

  static ImmutableCommunication_Payload.Builder builder() {
    return ImmutableCommunication_Payload.builder();
  }
}

File: medplum-typescript/src/model/fhir/Communication_Payload.ts
import { Attachment } from "./Attachment";
import { Extension } from "./Extension";
import { Reference } from "./Reference";

export interface Communication_Payload {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    contentString: string | null | undefined;
    contentAttachment: Attachment | null | undefined;
    contentReference: Reference | null | undefined;
}
